src/main/java/org/apache/sysml/hops/AggUnaryOp.java

```diff
@@ -463,6 +463,15 @@ else if ( getInput().get(0).areDimsBelowThreshold() || getInput().get(0).isVecto
 			//pull unary aggregate into spark 
 			_etype = ExecType.SPARK;
 		}
+
+		if( _etype == ExecType.CP && _etypeForced != ExecType.CP
+			&& !(getInput().get(0) instanceof DataOp)  //input is not checkpoint
+			&& getInput().get(0).getParent().size()==1 //uagg is only parent
+			&& getInput().get(0).optFindExecType() == ExecType.FLINK )
+		{
+			//pull unary aggregate into spark 
+			_etype = ExecType.FLINK;
```

Comment:
We could probably fuse these two predicates by checking inside the if-statement so that we don't have to copy this long predicate.